osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -377,63 +389,83 @@ private void handleEvent(string line)
                     break;
 
                 case LegacyEventType.Background:
-                    beatmap.BeatmapInfo.Metadata.BackgroundFile = CleanFilename(split[2]);
+                    split.MoveNext(); // split[1]
+                    split.MoveNext(); // split[2]
+                    beatmap.BeatmapInfo.Metadata.BackgroundFile = CleanFilename(split.CurrentSpan);
                     break;
 
                 case LegacyEventType.Break:
-                    double start = getOffsetTime(Parsing.ParseDouble(split[1]));
-                    double end = Math.Max(start, getOffsetTime(Parsing.ParseDouble(split[2])));
+                    split.MoveNext(); // split[1]
+                    double start = getOffsetTime(Parsing.ParseDouble(split.CurrentSpan));
+                    split.MoveNext(); // split[2]
+                    double end = Math.Max(start, getOffsetTime(Parsing.ParseDouble(split.CurrentSpan)));
 
                     beatmap.Breaks.Add(new BreakPeriod(start, end));
                     break;
             }
         }
 
-        private void handleTimingPoint(string line)
+        private void handleTimingPoint(ReadOnlySpan<char> line)
         {
-            string[] split = line.Split(',');
+            var split = line.Split(',');
 
-            double time = getOffsetTime(Parsing.ParseDouble(split[0].Trim()));
+            split.MoveNext(); // split[0]
+            double time = getOffsetTime(Parsing.ParseDouble(split.CurrentSpan));
```

Comment:
Note to self (and possibly other reviewers): `.Trim()` was dropped here, but that's okay, since `double.Parse()` seems to be [trimming whitespace anyway](https://learn.microsoft.com/en-us/dotnet/api/system.double.parse?view=net-7.0) (see remarks section).